api/handlers.go

```diff
@@ -161,6 +172,172 @@ func UnpackHandler(routePrefix string) http.HandlerFunc {
 	}
 }
 
+// SaveByFormHandler saves a dataset by reading the body from a file
+func SaveByFormHandler(inst *lib.Instance, routePrefix string) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		if r.Method != http.MethodPost {
+			util.NotFoundHandler(w, r)
+			return
+		}
+
+		if err := r.ParseMultipartForm(maxBodyFileSize); err != nil {
+			util.WriteErrResponse(w, http.StatusBadRequest, err)
+			return
+		}
+
+		p := &lib.SaveParams{}
+		if err := parseSaveParamsFromRequest(r, p); err != nil {
+			util.WriteErrResponse(w, http.StatusBadRequest, err)
+			return
+		}
+
+		if p.Dataset == nil {
+			p.Dataset = &dataset.Dataset{}
+		}
+		if err := formFileDataset(r, p.Dataset); err != nil {
+			util.WriteErrResponse(w, http.StatusBadRequest, err)
+			return
+		}
+
+		ds, err := inst.Dataset().Save(r.Context(), p)
+		if err != nil {
+			// what should the error response be
```

Comment:
I would suggest using `RespondWithError` from `api/util/errors.go`. It does some work to inspect the error and respond with an appropriate http status code. As long as `Dataset.Save` is returning known errors, the response code will be correct, and if it isn't, at least fixing that can be centralized to the `ResponseWithError` method.

Reply:
thank you I forgot this existed